src/core/SkColor.cpp

```diff
@@ -148,3 +151,31 @@ SkPMColor4f SkPMColor4f::FromBytes_RGBA(uint32_t c) {
     Sk4f_fromL32(c).store(&color);
     return color;
 }
+
+namespace SkSpotColors {
+
+std::unordered_map<SkString, SkColor4f> spotColors;
+
+std::vector<SkString> names() {
+    std::vector<SkString> names;
+    std::transform(spotColors.begin(), spotColors.end(), std::back_inserter(names), [](const auto& kv) { return kv.first; });
+    return names;
+}
+
+SkColor4f get(const SkString& name) {
+    if (const auto i = spotColors.find(name); i != spotColors.end()) {
+        return i->second;
+    }
+
+    return defaultRgbApproximation;
```

Comment:
Is it intended that Magenta is returned when the color is not found?

Reply:
Yes, because you don't want things to be invisible if a spot color is not yet defined. This follows the behavior of of templated strings where when loading a scene with templated strings, you have to set the values for the templates again because we don't save the values in the scene file. Until you set the values of the template strings, they show up as the template (e.g. `{{first_name}}`). Same thing here, when you load a scene, you must redefine the values of the spot colors. Until you do, they show up with a default color, magenta.

Now, I could've put (just) this piece of color fallback logic in our engine but I figured it was simpler that way, keeping all the spot color behavior logic in Skia.